### PR TITLE
feat(medusa, admin-ui): Improvements to product categories

### PR DESCRIPTION
**Pull request by riqwan**
What:
- Adds name as required in category create form
- Adds name and handle as required in category edit form
- Updates message on create/update forms
- Adds category indicators for is_internal and is_active fields in the tree list
- Fixes bug where tree is not reset when update fails

https://user-images.githubusercontent.com/5105988/223707500-db39da37-5a5c-4440-a39f-9b3240ea2316.mov

RESOLVES CORE-1202
RESOLVES CORE-1207
RESOLVES CORE-1201
RESOLVES CORE-1217

### CODE REVIEW

File: packages/admin-ui/ui/src/components/fundamentals/icons/tag-red-dot-icon/index.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+      {...attributes}
+    >
+      <circle cx="16" cy="16" r="6" fill="#FFE5E5" />
+      <circle cx="16" cy="16" r="3" fill="#E5484D" />
+    </svg>
+  )
+}
+
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/components/fundamentals/icons/tag-red-dot-icon/index.tsx:19]: suggested(non-blocking): Create a generic `TagStatusDotIcon` with a switch on colors.

```
const TagStatusDotIcon: React.FC<IconProps> = ({ size = "24", color = "some-color", ...attributes }) => {
  let circleOneFill = ""
  let circleTwoFill = ""
  switch (color):
     ...
  return (
```
Though, that would require that all dot icon SVGs be constructed using two circles. I don't know if that's the case. 

Also, this is very nit.

File: packages/admin-ui/ui/src/components/molecules/tooltip-icon/index.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from "react"
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/components/molecules/tooltip-icon/index.tsx:1]: question: Don't we have a tooltip icon already?

File: packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx [hunk 3]
```diff
@@ -53,15 +54,18 @@ function ProductCategoriesList(props: ProductCategoriesListProps) {
       }
 
       try {
+        isError && setIsError(false)
```
Comment (fPolic) [inline packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx:57]:
```suggestion
        setIsError(false)
```
won't rerender if already `false`

File: packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx [hunk 4]
```diff
@@ -26,6 +26,7 @@ function ProductCategoriesList(props: ProductCategoriesListProps) {
   const queryClient = useQueryClient()
   const notification = useNotification()
   const [isUpdating, setIsUpdating] = useState(false)
+  const [isError, setIsError] = useState(false)
```
Comment (fPolic) [inline packages/admin-ui/ui/src/domain/product-categories/components/product-categories-list.tsx:29]: suggestion: we have `useToggleState` hook which is nice util for boolean states

**Review by fPolic — APPROVED** [review]
LGTM 💪

**Pull request merged**